File: sample-conference-videochat/sample-conference-videochat/InfoScreen/InfoScreenViewController/InfoTableViewController.m
//
//  InfoTableViewController.m
//  sample-conference-videochat
//
//  Created by Injoit on 12/30/18.
//  Copyright © 2018 Quickblox. All rights reserved.
//

#import "InfoTableViewController.h"
#import "InfoModel.h"
#import "InfoTableViewCell.h"
#import <Quickblox/QBASession.h>

NSString *const kInfoTableViewCellId = @"InfoTableViewCell";
NSString *const kLogoTableViewCellId = @"QBLogoTableViewCell";


@interface InfoTableViewController ()
@property (strong, nonatomic) NSMutableArray<InfoModel*>* infoModels;
@end

@implementation InfoTableViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    UIBarButtonItem *backButtonItem = [[UIBarButtonItem alloc] initWithImage:[UIImage imageNamed:@"chevron"]
                                                                       style:UIBarButtonItemStylePlain
                                                                      target:self
                                                                      action:@selector(didTapBack:)];
    
    self.navigationItem.leftBarButtonItem = backButtonItem;
    backButtonItem.tintColor = UIColor.whiteColor;
    
    [self setupTableView];
}

#pragma mark - Actions
- (void)didTapBack:(UIButton *)sender {
    [self.navigationController popViewControllerAnimated:YES];
}

- (void)setupTableView
{
    self.infoModels = [NSMutableArray new];
    
    NSString *appName = NSBundle.mainBundle.infoDictionary[@"CFBundleName"];
    InfoModel *appNameModel = [[InfoModel alloc] init];
    appNameModel.title = @"Application name";
    appNameModel.info = [NSString stringWithFormat:@"%@", appName];
    [self.infoModels addObject:appNameModel];
    
    NSString *appVersion = NSBundle.mainBundle.infoDictionary[@"CFBundleShortVersionString"];
    InfoModel *appVersionModel = [[InfoModel alloc] init];
    appVersionModel.title = @"Application version";
    appVersionModel.info = [NSString stringWithFormat:@"%@", appVersion];
    [self.infoModels addObject:appVersionModel];
    
    InfoModel *quickBloxSdkVersionModel = [[InfoModel alloc] init];
    quickBloxSdkVersionModel.title = @"QuickBlox SDK version";
    quickBloxSdkVersionModel.info = [NSString stringWithFormat:@"%@", QuickbloxFrameworkVersion];
    [self.infoModels addObject:quickBloxSdkVersionModel];
    
    InfoModel *wbrtcVersionModel = [[InfoModel alloc] init];
    wbrtcVersionModel.title = @"QuickbloxWebRTC version";
    wbrtcVersionModel.info = [NSString stringWithFormat:@"%@", QuickbloxWebRTCFrameworkVersion];
    [self.infoModels addObject:wbrtcVersionModel];
    
    InfoModel *appIDModel = [[InfoModel alloc] init];
    appIDModel.title = @"Application ID";
    appIDModel.info = [NSString stringWithFormat:@"%@", @(QBSettings.applicationID)];
    [self.infoModels addObject:appIDModel];
    
    InfoModel *authKeyModel = [[InfoModel alloc] init];
    authKeyModel.title = @"Auhtorization key";
    authKeyModel.info = [NSString stringWithFormat:@"%@", QBSettings.authKey];
    [self.infoModels addObject:authKeyModel];
    
    InfoModel *authSecretModel = [[InfoModel alloc] init];
    authSecretModel.title = @"Auhtorization secret";
    authSecretModel.info = [NSString stringWithFormat:@"%@", QBSettings.authSecret];
    [self.infoModels addObject:authSecretModel];
    
    InfoModel *accountKeyModel = [[InfoModel alloc] init];
    accountKeyModel.title = @"Account key";
    accountKeyModel.info = [NSString stringWithFormat:@"%@", QBSettings.accountKey];
    [self.infoModels addObject:accountKeyModel];
    
    InfoModel *apiDomainModel = [[InfoModel alloc] init];
    apiDomainModel.title = @"API domain";
    apiDomainModel.info = [NSString stringWithFormat:@"%@", QBSettings.apiEndpoint];
    [self.infoModels addObject:apiDomainModel];
    
    InfoModel *chatDomainModel = [[InfoModel alloc] init];
    chatDomainModel.title = @"Chat domain";
    chatDomainModel.info = [NSString stringWithFormat:@"%@", QBSettings.chatEndpoint];
    [self.infoModels addObject:chatDomainModel];
    
    InfoModel *janusServerURLModel = [[InfoModel alloc] init];
    janusServerURLModel.title = @"Janus server url";
    janusServerURLModel.info = [NSString stringWithFormat:@"%@", QBRTCConfig.conferenceEndpoint];
    [self.infoModels addObject:janusServerURLModel];
    
    NSString *qaVersion = NSBundle.mainBundle.infoDictionary[@"CFBundleVersion"];
    InfoModel *qaVersionModel = [[InfoModel alloc] init];
    qaVersionModel.title = @"QA version";
    qaVersionModel.info = [NSString stringWithFormat:@"%@", qaVersion];
    [self.infoModels addObject:qaVersionModel];
    
    InfoModel *logoModel = [[InfoModel alloc] init];
    logoModel.title = @"logo";
    logoModel.info = @"logo";
    [self.infoModels addObject:logoModel];
    
    [self.tableView reloadData];
}

#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return self.infoModels.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    BOOL isLast = indexPath.row == self.infoModels.count - 1;
    if (isLast) {
        UITableViewCell* cell = [tableView dequeueReusableCellWithIdentifier:kLogoTableViewCellId];
        return cell;
    }
    InfoTableViewCell* cell = [tableView dequeueReusableCellWithIdentifier:kInfoTableViewCellId];
    InfoModel *model = self.infoModels[indexPath.row];
    [cell applyInfo:model];
    return cell;
}

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath
{
    BOOL isLast = indexPath.row == self.infoModels.count - 1;
    return isLast ? 80.0f : 54.0f;
}

@end

File: sample-videochat-webrtc-swift/sample-videochat-webrtc-swift/Components/InfoScreenViewController/InfoTableViewController.swift
//
//  InfoTableViewController.swift
//  sample-videochat-webrtc-swift
//
//  Created by Injoit on 12/30/18.
//  Copyright © 2018 QuickBlox. All rights reserved.
//

import UIKit
import Quickblox
import QuickbloxWebRTC

struct InfoCellConstant {
    static let infoTableViewCellId = "InfoTableViewCell"
    static let logoTableViewCellId = "QBLogoTableViewCell"
}

class InfoTableViewController: UITableViewController {
    
    var infoModels = [InfoModel]()
    override func viewDidLoad() {
        super.viewDidLoad()
        
        setupTableView()
    }
    
    private func setupTableView() {
        let appName = Bundle.main.infoDictionary?["CFBundleName"] as? String
        let appNameModel = InfoModel()
        appNameModel.title = "Application name:"
        appNameModel.info = appName ?? ""
        infoModels.append(appNameModel)
        
        let appVersion = Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String
        let appVersionModel = InfoModel()
        appVersionModel.title = "Application version:"
        appVersionModel.info = appVersion ?? ""
        infoModels.append(appVersionModel)
        
        let quickBloxSdkVersionModel = InfoModel()
        quickBloxSdkVersionModel.title = "QuickBlox SDK version:"
        quickBloxSdkVersionModel.info = QuickbloxFrameworkVersion
        infoModels.append(quickBloxSdkVersionModel)
        
        let wbrtcVersionModel = InfoModel()
        wbrtcVersionModel.title = "QuickbloxWebRTC version:"
        wbrtcVersionModel.info = QuickbloxWebRTCFrameworkVersion
        infoModels.append(wbrtcVersionModel)
        
        let appIDModel = InfoModel()
        appIDModel.title = "Application ID:"
        appIDModel.info = "\(QBSettings.applicationID)"
        infoModels.append(appIDModel)
        
        let authKeyModel = InfoModel()
        authKeyModel.title = "Auhtorization key:"
        authKeyModel.info = QBSettings.authKey ?? ""
        infoModels.append(authKeyModel)
        
        let authSecretModel = InfoModel()
        authSecretModel.title = "Auhtorization secret:"
        authSecretModel.info = QBSettings.authSecret ?? ""
        infoModels.append(authSecretModel)
        
        let accountKeyModel = InfoModel()
        accountKeyModel.title = "Account key:"
        accountKeyModel.info = QBSettings.accountKey ?? ""
        infoModels.append(accountKeyModel)
        
        let apiDomainModel = InfoModel()
        apiDomainModel.title = "API domain:"
        apiDomainModel.info = QBSettings.apiEndpoint ?? ""
        infoModels.append(apiDomainModel)
        
        let chatDomainModel = InfoModel()
        chatDomainModel.title = "Chat domain:"
        chatDomainModel.info = QBSettings.chatEndpoint ?? ""
        infoModels.append(chatDomainModel)
        
        let logoModel = InfoModel()
        logoModel.title = "logo"
        logoModel.info = "logo"
        infoModels.append(logoModel)
        
        tableView.reloadData()
    }
    
    // MARK: - Table view data source
    
    override func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return infoModels.count
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let isLast = indexPath.row == infoModels.count - 1
        if isLast == true {
            let cell = tableView.dequeueReusableCell(withIdentifier: InfoCellConstant.logoTableViewCellId, for: indexPath)
            return cell
        }
        guard let cell = tableView.dequeueReusableCell(withIdentifier: InfoCellConstant.infoTableViewCellId)
            as? InfoTableViewCell else {
                return UITableViewCell()
        }
        let model = infoModels[indexPath.row]
        cell.applyInfo(model: model)
        return cell
    }
    
    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        let isLast = indexPath.row == infoModels.count - 1
        return isLast ? 80.0 : 54.0
    }
}
